### PR TITLE
fixes #12954 - config status is relevant if there are reports

### PR DESCRIPTION
**Pull request by domcleal**
For a host that has config management reports, but no Puppet proxy set,
consider the config status relevant.  This will have the side effect of
considering the status still relevant even if a host has a proxy unset
and old reports, but helps in an environment with reports but no proxy.

Depending on the presence of a report in a no-proxy environment does
mean that the "no reports" status probably wouldn't ever be set, as the
status wouldn't yet be relevant.

### CODE REVIEW

File: app/models/host_status/configuration_status.rb [hunk 1]
```diff
@@ -78,7 +78,7 @@ def to_status(options = {})
     end
 
     def relevant?
-      host.configuration?
+      host.configuration? || last_report.present?
```
Comment (ohadlevy) [inline app/models/host_status/configuration_status.rb]: so this means new host without a last_report will never show up its configuration status without a proxy?

Comment (domcleal) [inline app/models/host_status/configuration_status.rb]: Correct.

Comment (ohadlevy) [inline app/models/host_status/configuration_status.rb]: Maybe we could expose a setting? as a way saying I don't care about proxies? I have no idea how common this usage case is.

Comment (domcleal) [inline app/models/host_status/configuration_status.rb]: Can do, I've pushed another commit to add a setting.  It works slightly differently to 1.9 still, as the no reports status means this shows as a warning state for the host, rather than the greyed out state previously.

Comment (ares) [inline app/models/host_status/configuration_status.rb]: That makes sense to me, if I explicitly enable always_show_configuration_status, I guess that warning is what I want to see if there's no report. The warning basically says that we are waiting for report to be uploaded.

I don't want to complicate this more so take it just as suggestion if users complain. Maybe we could take Puppet interval setting, add it to `created_at` time of host (if it's build == false) and display warning only if it's past this window. This wouldn't solve it for rebuilds though, we'd need built_at attribute.

File: test/unit/domain_test.rb [hunk 2]
```diff
@@ -167,8 +167,17 @@ def create_a_host
     assert Domain.first.nameservers.empty?
   end
 
-  test "should query remote nameservers" do
-    assert Domain.first.nameservers.empty?
+  test "should query remote nameservers from domain SOA" do
```
Comment (domcleal) [inline test/unit/domain_test.rb:170]: This seemingly unrelated change is because of the test_helper fix, which stops setting changes leaking between tests (previously only in integration tests).  This test relied on the behaviour of the previous test.

**Comment by ohadlevy** [comment]
[test]

**Comment by ohadlevy** [comment]
@domcleal I assume you intend to keep the two separate commits?

**Comment by domcleal** [comment]
No, I didn't, it was to make reviewing easier.  Squashed.

**Comment by ohadlevy** [comment]
merged as 8660f93 thanks @domcleal
